dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.TextToImage.cs

```diff
@@ -47,4 +49,113 @@ internal async Task<string> GenerateImageAsync(
 
         return generatedImage.ImageUri?.ToString() ?? throw new KernelException("The generated image is not in url format");
     }
+
+    /// <summary>
+    /// Generates an image with the provided configuration.
+    /// </summary>
+    /// <param name="input">The input text content to generate the image</param>
+    /// <param name="executionSettings">Execution settings for the image generation</param>
+    /// <param name="kernel">Kernel instance</param>
+    /// <param name="cancellationToken">Cancellation token</param>
+    /// <returns>List of image generated contents</returns>
+    internal async Task<IReadOnlyList<ImageContent>> GetImageContentsAsync(
+        TextContent input,
+        PromptExecutionSettings? executionSettings = null,
+        Kernel? kernel = null,
+        CancellationToken cancellationToken = default)
+    {
+        // Ensure the input is valid
+        Verify.NotNull(input);
+
+        // Convert the generic execution settings to OpenAI-specific settings
+        var imageSettings = OpenAITextToImageExecutionSettings.FromExecutionSettings(executionSettings);
+
+        var imageGenerationOptions = new ImageGenerationOptions()
+        {
+            Size = GetGeneratedImageSize(imageSettings.Size),
+            ResponseFormat = GetResponseFormat(imageSettings.ResponseFormat),
+            Style = GetGeneratedImageStyle(imageSettings.Style),
+            Quality = GetGeneratedImageQuality(imageSettings.Quality),
+            EndUserId = imageSettings.EndUserId,
+        };
+
+        var targetModel = string.IsNullOrEmpty(imageSettings.ModelId)
+            ? "dall-e-2" // Defaults to the DALL-E 2 server-side - https://platform.openai.com/docs/api-reference/images/create#images-create-model.
+            : imageSettings.ModelId;
+
+        ClientResult<GeneratedImage> response = await RunRequestAsync(() => this.Client!.GetImageClient(targetModel).GenerateImageAsync(input.Text, imageGenerationOptions, cancellationToken)).ConfigureAwait(false);
+        var generatedImage = response.Value;
+
+        List<ImageContent> result = [];
+        if (generatedImage.ImageUri is not null)
+        {
+            result.Add(new ImageContent(uri: generatedImage.ImageUri) { InnerContent = generatedImage });
+        }
+        else
+        {
+            result.Add(new ImageContent(generatedImage.ImageBytes, "image/png") { InnerContent = generatedImage });
```

Comment:
I wonder if we can get content type from AI model instead of hardcoding it.

Reply:
We can't, they don´t provide what is the type anywhere in the documentation, I ended up finding out it was PNG by checking the header in the binary format sent back.